ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/util/BlobSidecarPoolImpl.java

```diff
@@ -0,0 +1,369 @@
+  private final RecentChainData recentChainData;
+  private final int maxTrackers;
+
+  private final Function<SlotAndBlockRoot, BlockBlobSidecarsTracker> trackerFactory;
+
+  private final Subscribers<RequiredBlockRootSubscriber> requiredBlockRootSubscribers =
+      Subscribers.create(true);
+  private final Subscribers<RequiredBlockRootDroppedSubscriber>
+      requiredBlockRootDroppedSubscribers = Subscribers.create(true);
+
+  private final Subscribers<RequiredBlobSidecarSubscriber> requiredBlobSidecarSubscribers =
+      Subscribers.create(true);
+  private final Subscribers<RequiredBlobSidecarDroppedSubscriber>
+      requiredBlobSidecarDroppedSubscribers = Subscribers.create(true);
+
+  private int totalBlobSidecars;
+
+  BlobSidecarPoolImpl(
+      final SettableLabelledGauge sizeGauge,
+      final Spec spec,
+      final TimeProvider timeProvider,
+      final AsyncRunner asyncRunner,
+      final RecentChainData recentChainData,
+      final UInt64 historicalSlotTolerance,
+      final UInt64 futureSlotTolerance,
+      final int maxTrackers) {
+    super(spec, futureSlotTolerance, historicalSlotTolerance);
+    this.spec = spec;
+    this.timeProvider = timeProvider;
+    this.asyncRunner = asyncRunner;
+    this.recentChainData = recentChainData;
+    this.maxTrackers = maxTrackers;
+    this.sizeGauge = sizeGauge;
+    this.trackerFactory = BlockBlobSidecarsTracker::new;
+
+    // Init the label so it appears in metrics immediately
+    sizeGauge.set(0, GAUGE_BLOB_SIDECARS_LABEL);
+    sizeGauge.set(0, GAUGE_BLOB_SIDECARS_TRACKERS_LABEL);
+  }
+
+  @VisibleForTesting
+  BlobSidecarPoolImpl(
+      final SettableLabelledGauge sizeGauge,
+      final Spec spec,
+      final TimeProvider timeProvider,
+      final AsyncRunner asyncRunner,
+      final RecentChainData recentChainData,
+      final UInt64 historicalSlotTolerance,
+      final UInt64 futureSlotTolerance,
+      final int maxTrackers,
+      final Function<SlotAndBlockRoot, BlockBlobSidecarsTracker> trackerFactory) {
+    super(spec, futureSlotTolerance, historicalSlotTolerance);
+    this.spec = spec;
+    this.timeProvider = timeProvider;
+    this.asyncRunner = asyncRunner;
+    this.recentChainData = recentChainData;
+    this.maxTrackers = maxTrackers;
+    this.sizeGauge = sizeGauge;
+    this.trackerFactory = trackerFactory;
+
+    // Init the label so it appears in metrics immediately
+    sizeGauge.set(0, GAUGE_BLOB_SIDECARS_LABEL);
+    sizeGauge.set(0, GAUGE_BLOB_SIDECARS_TRACKERS_LABEL);
+  }
+
+  @Override
+  public synchronized void onNewBlobSidecar(final BlobSidecar blobSidecar, final boolean syncing) {
+    if (shouldIgnoreItemAtSlot(blobSidecar.getSlot())) {
+      return;
+    }
+
+    final SlotAndBlockRoot slotAndBlockRoot = blobSidecar.getSlotAndBlockRoot();
+
+    final BlockBlobSidecarsTracker blobSidecarsTracker =
+        getOrCreateBlobSidecarsTracker(
+            slotAndBlockRoot,
+            newTracker -> {
+              if (!syncing) {
```

Comment:
it is currently intentional but, as I mention in the PR description, it may go away at the end (just let the `firstSeen` checks go, and will not trigger any fetch by design since, by definition, when we sync we are dealing with old slots)